rework_pysatl_mpest/core/mixture.py

```diff
@@ -72,31 +77,43 @@ class MixtureModel:
         generate
     """
 
-    def __init__(self, components: Sequence["ContinuousDistribution"], weights: Optional[ArrayLike] = None):
+    _dtype: type[DType]
+
+    def __init__(
+        self,
+        components: Sequence["ContinuousDistribution[DType]"],
+        weights: Optional[ArrayLike] = None,
+        dtype: type[DType] | None = None,
+    ):
         n_components = len(components)
         if n_components == 0:
             raise ValueError("List of components cannot be an empty")
 
+        if dtype is None:
+            self._dtype = np.float64  # type: ignore[assignment]
+        else:
+            self._dtype = dtype
+
         if weights is None:
             weights = np.full(n_components, 1.0 / n_components)
         else:
-            weights = np.asarray(weights, dtype=float64)
+            weights = np.asarray(weights, dtype=self.dtype)
             self._validate_weights(n_components, weights)
 
         self._components = list(components)
         self._log_weights = np.log(weights + 1e-30)
```

Comment:
There is the constant 1e-30 is Python float number, so:
```suggestion
        self._log_weights = np.log(weights + self.dtype(1e-30))
```